### PR TITLE
fix

### PR DESCRIPTION
**Pull request by GusCaplan**

### CODE REVIEW

**Comment by hydrabolt** [comment]
wow what a descriptive name for this PR

File: src/structures/MessageEmbed.js [hunk 1]
```diff
@@ -49,6 +49,12 @@ class MessageEmbed {
      * @type {Object}
      */
     this.footer = data.footer;
+    
+    /**
+     * The fields of the embed
```
Comment (iCrawl) [inline src/structures/MessageEmbed.js]: should be `this` embed for consistency

File: src/structures/MessageEmbed.js [hunk 2]
```diff
@@ -61,6 +73,12 @@ class MessageEmbed {
      * @type {MessageEmbedProvider}
      */
     this.provider = data.provider ? new MessageEmbedProvider(this, data.provider) : null;
+
+    /**
+     * The footer of this embed
+     * @type {MessageEmbedFooter}
+     */
+    this.footer = new MessageEmbedFooter(data.footer);
```
Comment (HyperCoder2975) [inline src/structures/MessageEmbed.js]: Should be `this.footer = data.footer ? new MessageEmbedFooter(this, data.footer) : null;`

File: src/structures/MessageEmbed.js [hunk 3]
```diff
@@ -45,6 +45,19 @@ class MessageEmbed {
     this.url = data.url;
 
     /**
+     * The fields of this embed
+     * @type {Array<MessageEmbedField>}
+     */
+    this.fields = [];
+    for (const field of data.fields) this.fields.push(new MessageEmbedField(embed, field));
```
Comment (HyperCoder2975) [inline src/structures/MessageEmbed.js]: If data.fields is null, wouldn't this cause an error?

**Pull request merged**